google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java

```diff
@@ -1721,10 +1722,12 @@ public void testSessionMetrics() throws Exception {
     latch.await();
     // Wait until the request has timed out.
     int waitCount = 0;
-    while (pool.getNumWaiterTimeouts() == 0L && waitCount < 1000) {
-      Thread.sleep(5L);
+    while (pool.getNumWaiterTimeouts() == 0L && waitCount < 5000) {
```

Comment:
There is similar waits added in https://github.com/googleapis/java-spanner/blob/5ee5540ac580b75bae65dce9407bbf43c97855f6/google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java#L1169
and
https://github.com/googleapis/java-spanner/blob/5ee5540ac580b75bae65dce9407bbf43c97855f6/google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java#L1217.

Does this change apply there also?

Reply:
Those tests are not affected by the same problem as the test that this change is for. The 'testSessionMetrics` test had two problems:
1. It did not actually check for what it should test, as the timeout that it was supposed to verify was disabled.
2. The verify calls did not actually check that the thing that was supposed to happen, actually happened.
3. And as it was not triggering the condition it should, it waited the full 5 seconds, before finishing without checking anything.

The other tests do check what they should, and are also not waiting the full 5 seconds. But I agree that we should be consistent in the way we execute these waits, so I updated the wait loops to sleep for 1ms instead of 5ms.

Reply:
Thanks for explaining this.